src/uwtools/config_validator.py

```diff
@@ -2,61 +2,70 @@
 Support for validating a config using JSON Schema.
 """
 import json
-import os
+from pathlib import Path
+from typing import List
 
 import jsonschema
 
 from uwtools.config import YAMLConfig
 from uwtools.logger import Logger
 
+# Public
 
-def config_is_valid(config_file: str, validation_schema: str, log: Logger) -> bool:
+
+def config_is_valid(config_file: str, schema_file: str, log: Logger) -> bool:
     """
-    Validate a config using JSON Schema.
+    Check whether the given config file conforms to the given JSON Schema spec and whether any
+    filesystem paths it identifies do not exist.
     """
-    # Get the config file to be validated and dereference Jinja2 templates.
-    # The config file will only be 2 levels deep.
-
-    config_obj = YAMLConfig(config_file, log_name=log.name)
-    config_obj.dereference_all()
-
-    # Load the JSON Schema validation schema.
-
-    with open(validation_schema, "r", encoding="utf-8") as schema_file:
-        schema = json.load(schema_file)
-
-    # Validate the config file against the schema file.
-
-    schema_error = 0
-    validator = jsonschema.Draft7Validator(schema)
-
-    # Print out each schema error.
-
-    errors = validator.iter_errors(config_obj.data)
+    # Load the config and schema.
+    yaml_config = YAMLConfig(config_file, log_name=log.name)
+    yaml_config.dereference_all()
+    with open(schema_file, "r", encoding="utf-8") as f:
+        schema = json.load(f)
+    # Collect and report on schema-validation errors.
+    errors = _validation_errors(yaml_config.data, schema)
+    log_method = log.error if errors else log.info
+    log_method("%s schema-validation error%s found", len(errors), "" if len(errors) == 1 else "s")
     for error in errors:
-        schema_error += 1
         log.error(error)
         log.error("------")
+    # It's pointless to evaluate an invalid config, so return now if that's the case.
+    if errors:
+        return False
+    # Collect and report bad paths found in config.
+    if bad_paths := _bad_paths(yaml_config.data, schema):
+        for bad_path in bad_paths:
+            log.error("Path does not exist: %s", bad_path)
+        return False
+    # If no issues were detected, report success.
+    return True
 
-    # Create a list of fields that could contain a file or path.
 
-    path_list = []
-    for field in schema["properties"]:
-        for value in schema["properties"][field]["properties"]:
-            # the json pair "format": "uri" labels a path or file
-            if "format" in schema["properties"][field]["properties"][value]:
-                path_list.append(value)
+# Private
 
-    # Check for existence of those files or paths.
 
-    for field in config_obj.data:
-        for value in config_obj.data[field]:
```

Comment:
Here's where a `str` value of `config_obj.data[field]` would result in iteration over the individual letters of the string.